### ISSUE TITLE
Format rule request: Remove all semicolons at the end of lines

### ISSUE THREAD

**Issue by adbertram** [PowerShell/vscode-powershell#801]
I know some people love their semi-colons at the end of lines like in C# but it's not necessary for PowerShell. I'd love to take someone else's code and format the document to remove all of the semicolons from it.

**Comment by pcgeek86** [comment]
Maybe this is better suited for PSScriptAnalyzer?

**Comment by daviwil** [comment]
Part of the implementation would happen in the PowerShell extension so I'd say we can leave the issue here for tracking purposes.

**Comment by rkeithhill** [comment]
This could be make be a good example of using $psEditor.  :-)

**Comment by TylerLeonhardt** [comment]
@bergmeister I feel like this can all be handled in PSSA via Formatting. What do you think?

**Comment by bergmeister** [comment]
Yes, there is already an open issue: https://github.com/PowerShell/PSScriptAnalyzer/issues/824
It should be a small to medium sized task (half a day for someone who knows a bit about PSSA and maybe a full day for a newbie) but personally, I do not see much value and more a 'nice to have', therefore it's up for grabs :-)

**Comment by SydneyhSmith** [comment]
@bergmeister thanks for the response and for linking the issue where this is being tracked, I will go ahead and leave this open as way to track the work that would need to happen to allow this to be a configuration in vscode

**Comment by SeeminglyScience** [comment]
Anyone who wants to pick this up can use [Remove-Semicolon](https://github.com/SeeminglyScience/EditorServicesCommandSuite/blob/7c2089565af192c074d5b1ecd65cd583f869cee6/module/Public/Remove-Semicolon.ps1) from EditorServicesCommandSuite as guide.  Implementing it in PSSA won't be as simple, but that should give an idea of what you're looking for.
